### ISSUE TITLE
TypeError: Cannot read property 'first' of undefined

### ISSUE THREAD

**Issue by gabrieloliverio** [ratiw/vuetable-2#133]
Hello Ratiw!

I'm getting the following error when I try to put a `vuetable-2` version 1.6.0 in my component:

```
[Vue warn]: Error in render function: "TypeError: Cannot read property 'first' of undefined"

found in

---> <VuetablePagination> at /var/www/html/sicorp/frontend/node_modules/vuetable-2/src/components/VuetablePagination.vue
       <ConsultarVara> at /var/www/html/sicorp/frontend/src/components/acao/vara/index.vue
         <App> at /var/www/html/sicorp/frontend/src/App.vue
           <Root>
```

Source:

```
<div class="panel panel-default">
              <div class="panel-heading">
                <h3 class="panel-title">Title</h3>
              </div>
              <div class="panel-body">
                  <div class="btn-group pull-left" role="group">
                  </div>

                  <vuetable ref="vuetable"
                      :api-url="apiUrl"
                      :fields="[
                        {
                          name: 'field1'
                        },
                        {
                          name: 'field2'
                        }
                      ]"
                      :css="css.table"
                      data-path="data"
                      pagination-path=""
                      :append-params="myObject"
                      @vuetable:pagination-data="onPaginationData">
                  </vuetable>
              </div>
              <div class="panel-footer">
                  <vuetable-pagination-info ref="paginationInfo"
                      :css="css.paginationInfo"
                      info-class="pull-left">
                  </vuetable-pagination-info>
                  <vuetable-pagination ref="pagination"
                      :css="css.pagination"
                      :icons="css.icons"
                      @vuetable-pagination:change-page="onChangePage">
                  </vuetable-pagination>
              </div>
          </div>
```

I register the `Vuetable`, `VuetablePagination` and `VuetablePaginationInfo`:

```
components: {
        'vuetable': Vuetable,
        'vuetable-pagination': VuetablePagination,
        'vuetable-pagination-info': VuetablePaginationInfo,
}
```

And define the methods:

```
methods: {
        onPaginationData (paginationData) {
            this.$refs.pagination.setPaginationData(paginationData)
            this.$refs.paginationInfo.setPaginationData(paginationData)
        },
        onChangePage(page) {
            this.$refs.vuetable.changePage(page)
        }
    },
```

What am I doing wrong?

**Comment by ratiw** [comment]
Please see the Breaking Change [here](https://github.com/ratiw/vuetable-2#v160)

**Comment by gabrieloliverio** [comment]
Thanks @ratiw!

**Comment by jdriesen** [comment]
Hi @ratiw 
[Indeed, this solution solved the "**_Cannot read property 'first' of undefined_**" problem](https://codepen.io/ratiw/pen/GmJayw).
Though, if I can, I would recommend to make a small modification to this solution.
(I struggled a bit with it ...)

Currently, your data property is returning an object.
```
data: {
... }

```
The new Vue Syntax is supposing the dataobject is a function:
Code should be smt like:

```
data() {
  return {
  ... 
  }
}

```
I hope I saved some time for other developers.

Kind regards,
Johnny (Belgium)
